frontik/async.py

```diff
@@ -95,3 +96,51 @@ def new_cb(*args, **kwargs):
             self.try_finish()
 
         return new_cb
+
+
+def future_fold(future, map_func=None, exception_func=None):
+    """
+    Process sucessed result of the future with map_func.
+    Process failed result with error_func.
+
+    Functions may raise an Exception - it will sent to the result future as exception - or
```

Comment:
Тут достаточно тогда просто развернуть что происходит при эксепшене:
If `result_mapper` or `exception_mapper` raises an exception, it will be set as an exception for the resulting future.